src/main/java/thePackmaster/potions/PackInAJar.java

```diff
@@ -0,0 +1,82 @@
+package thePackmaster.potions;
+
+
+import basemod.abstracts.CustomPotion;
+import com.badlogic.gdx.graphics.Color;
+import com.megacrit.cardcrawl.actions.common.DiscardAction;
+import com.megacrit.cardcrawl.actions.common.MakeTempCardInHandAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.helpers.PowerTip;
+import com.megacrit.cardcrawl.localization.PotionStrings;
+import thePackmaster.SpireAnniversary5Mod;
+import thePackmaster.actions.EasyModalChoiceAction;
+import thePackmaster.cards.PackmasterModalChoiceCard;
+import thePackmaster.packs.AbstractCardPack;
+import thePackmaster.util.Wiz;
+
+import java.util.ArrayList;
+
+import static thePackmaster.SpireAnniversary5Mod.makeID;
+
+
+public class PackInAJar extends CustomPotion {
+    public static final String POTION_ID = SpireAnniversary5Mod.makeID("PackInAJar");
+    private static final PotionStrings potionStrings = CardCrawlGame.languagePack.getPotionString(POTION_ID);
+    public static final String NAME = potionStrings.NAME;
+    public static final String[] DESCRIPTIONS = potionStrings.DESCRIPTIONS;
+
+    public PackInAJar() {
+        super(NAME, POTION_ID, PotionRarity.UNCOMMON, PotionSize.GHOST, PotionColor.NONE);
+        this.isThrown = false;
+        this.targetRequired = false;
+        this.labOutlineColor = Color.TAN.cpy();
+        this.tips.clear();
+        this.tips.add(new PowerTip(this.name, this.description));
+    }
+
+
+    public void initializeData() {
+        this.potency = getPotency();
+        if (potency > 1){
+            this.description = potionStrings.DESCRIPTIONS[0] + potency + potionStrings.DESCRIPTIONS[2];
+        } else {
+            this.description = potionStrings.DESCRIPTIONS[0] + potency + potionStrings.DESCRIPTIONS[1];
+        }
+    }
+
+    public void use(AbstractCreature target) {
+        ArrayList<AbstractCardPack> choices = SpireAnniversary5Mod.getRandomPacks(false, 3);
+
+        ArrayList<AbstractCard> packCards = new ArrayList<>();
+
+        for (AbstractCardPack pack : choices) {
+            packCards.add(new PackmasterModalChoiceCard(pack.previewPackCard.cardID, pack.previewPackCard.name, pack.previewPackCard.rawDescription, true, () -> action(pack)));
+        }
+
+        addToBot(new EasyModalChoiceAction(packCards, 1, CardCrawlGame.languagePack.getUIString(makeID("ModalChoice")).TEXT[0]));
+    }
+
+    public void action(AbstractCardPack pack) {
+        addToBot(new DiscardAction(AbstractDungeon.player, AbstractDungeon.player, Wiz.hand().size(), true, false));
+        for (int i = 0; i < potency; i++) {
+            for (AbstractCard c : pack.cards) {
+                if (c.rarity == AbstractCard.CardRarity.COMMON || c.rarity == AbstractCard.CardRarity.UNCOMMON || c.rarity == AbstractCard.CardRarity.RARE) {
```

Comment:
We should also filter out cards that have the `CardTags.HEALING` tag, since those should never be generated in combat.

Reply:
Hm... fundamentally I would agree, but this makes the theming of the potion invalid.  It's the revival of Pack Rip, which would feel odd if it sometimes can give 10 cards and sometimes 8 because of HEALING.  I would say this is a case where it should be an exception to the rule.

Reply:
I understand your point, which means we have a dilemma. Happily, I have a suggestion that can square this circle.

To accept that this potion can give healing tagged cards, we just need for the potion itself to be given the same treatment as Fruit Juice and Smithing Oil -- no creating it from in-combat potion generation. That will resolve any issues I have with this, because it will prevent the thing that can provide permanent benefit (in this case, the Pack in a Jar potion) from being generated during combat.

Reply:
Good call, I have made it so!